src/routes/chat/Chat.js

```diff
@@ -12,32 +11,56 @@ import {
   chatAddMessage,
   chatMessageIsTyping,
   fetchMessages,
+  updateMessage,
+  setCurrentMessage,
+  removeMessage,
 } from '../../actions/security-chat';
 import socket from '../../utils/socket';
 // Without api need to use testData instead messages in <Messages> component
 // import testData from './testData.json';
 
 import s from './Chat.css';
 
-const testUserId = '1';
+const testUserId = 1;
 
 function Chat() {
   useStyles(s);
   const dispatch = useDispatch();
   const messages = useSelector(property('chat.messages'));
+  const message = useSelector(property('chat.message'));
+  const messageId = useSelector(property('chat.messageId'));
   useEffect(() => {
     dispatch(fetchMessages());
     socket.on('SERVER:NEW_MESSAGE', function reloadMessagesList() {
       dispatch(fetchMessages());
     });
   }, [dispatch]);
+  useEffect(() => {
+    socket.on('SERVER:REMOVE_MESSAGE', function reloadMessagesList() {
+      dispatch(fetchMessages());
+    });
+  }, [dispatch]);
+  useEffect(() => {
+    socket.on('SERVER:UPDATE_MESSAGE', function reloadMessagesList() {
+      dispatch(fetchMessages());
+    });
+    return function cleanupListener() {
+      console.info('removeEL message ');
+      window.removeEventListener('message', message);
```

Comment:
It seems that you remove not the listener you attached, don't you?

Reply:
I looked at the examples and update listeners, but
not completely sure that is right. Please tell me your if it`s right.

Reply:
try this https://stackoverflow.com/a/34716449/11621840